### PR TITLE
[#2251] Fix date logic issue (attempt 2)

### LINKED ISSUE #2251: Logic with "now" variable not triggering with "bigger than" and "less than"
### Product versie / Product version

latest

### Omschrijf het probleem / Describe the bug

Using the "now" (datetime) variable as a trigger the bigger/less than don't work as expected. 

This trigger does not work, today is 10/26/2022:
image

The trigger does work with equal and not equal values.

### Stappen om te reproduceren / Steps to reproduce

_No response_

### Verwacht gedrag / Expected behavior

_No response_

### Screen resolution

_No response_

### Device

_No response_

### OS

_No response_

### Browser

_No response_

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #2251 

This is mostly a bandaid fix while we wait for https://github.com/open-formulieren/open-forms/issues/2324

Changes in JSON logic for this PR: https://github.com/maykinmedia/json-logic-py/pull/10

This adds the variable `today` which was also requested here: #2357

### CODE REVIEW

File: src/openforms/js/components/admin/forms/Inputs.js [hunk 1]
```diff
@@ -28,6 +29,48 @@ const NumberInput = props => <Input type="number" {...props} />;
...
+      ...extraProps,
+    });
+  }, []);
+
+  const placeHolder =
+    extraProps.placeholder ||
+    defineMessage({
+      description: 'datetime input widget placeholder',
+      defaultMessage: 'Select date/time...',
+    });
```
Comment (sergei-maertens) [inline src/openforms/js/components/admin/forms/Inputs.js:58]: this means you can't simply pass an already `FormattedMessage` or string to `placeholder`, as it's passed down to `intl.formatMessage`. Maybe it'd be better to use `intl.formatMessage` directly in the default value and then just forward `placeholder`:

```jsx
const placeholder = extraProps.placeholder ?? intl.formatMesage({description: ..., defaultMessage: ...});
...
return (
    <input ref={...} placeholder={placeholder} />
)
```

File: src/openforms/variables/static_variables/static_variables.py [hunk 2]
```diff
@@ -20,6 +20,15 @@ def get_initial_value(self, submission: Optional[Submission] = None):
         return timezone.now()
 
 
+@register_static_variable("today")
+class Today(BaseStaticVariable):
+    name = _("Today")
+    data_type = FormVariableDataTypes.date
+
+    def get_initial_value(self, submission: Optional[Submission] = None) -> str:
+        return timezone.now().date()
```
Comment (sergei-maertens) [inline src/openforms/variables/static_variables/static_variables.py:29]: `timezone.now()` returns a datetime with timezone information in UTC, which means if you execute this at Nov 24th 00:30 AM, the reported date will actually be Nov 23rd -> please add a unit test for that :-)

The correct way would be:

```py
now_utc = timezone.now()
timezone.localtime(now_utc).date()
```

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/js/components/admin/forms/Inputs.js [hunk 3]
```diff
@@ -28,6 +29,48 @@ const NumberInput = props => <Input type="number" {...props} />;
...
+      target: {name: name, value: formattedDatetime},
+    };
+    onChange(fakeEvent);
+  };
+
+  useEffect(() => {
+    flatpickr(datetimePickerRef.current, {
+      onChange: wrappedOnChange,
+      ...extraProps,
+    });
```
[the remark this answers is not in the document]
Comment (SilviaAmAm) [reply src/openforms/js/components/admin/forms/Inputs.js:50]: Ahh yes, I had made a mental note of adding the destroy call, but clearly it should have been a physical note :D

File: src/openforms/variables/static_variables/static_variables.py (hunk 2 above)
Comment (SilviaAmAm) [reply src/openforms/variables/static_variables/static_variables.py]: :see_no_evil:

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**